source/MaterialXCore/Definition.cpp

```diff
@@ -126,11 +125,11 @@ bool NodeDef::validate(string* message) const
 
 bool NodeDef::isVersionCompatible(ConstElementPtr elem) const
 {
-    if (getVersionIntegers() == elem->getVersionIntegers())
+    if (getVersionString() == elem->getVersionString())
```

Comment:
Fix to compare actual string as version number utility can fail in the general string version case and only handles 
versions in the format "<x>.<y>".